_posts/2019-10-02-#16401.md

```diff
@@ -8,7 +8,76 @@ host: jonatack
+long-time Bitcoin Core contributor and co-founder of [Chaincode
+Labs](https://chaincode.com), who [has been
+working](https://github.com/bitcoin/bitcoin/pulls?q=is%3Apr+author%3Asdaftuar)
+to improve the privacy and resilience of Bitcoin's peer-to-peer network.
+
+Suhas described today's PR as a proof-of-concept motivation for refactoring
+AcceptToMemoryPoolWorker (ATMP), which he did in [PR
+#16400](https://github.com/bitcoin/bitcoin/pull/16400) "Rewrite
+AcceptToMemoryPoolWorker() using smaller parts" that was merged two weeks ago
+and which lays a foundation for multiple transactions for package relay.
+
+Today's PR was opened at the same time as the ATMP refactoring, thus before it
+was merged, and requires no additional p2p protocol changes.
+
+Suhas' motivation for it begins: *Accepting a single transaction to the mempool
+only succeeds if (among other things) the feerate of the transaction is greater
+than both the min relay fee and the mempool min fee. Consequently, a transaction
+below the minimum fee may not be accepted to the mempool, even if we later learn
+of a transaction with a high fee that depends on it.*
+
+- You can find the minimum relay fee and mempool minimum fee in the codebase by
+git grepping for CheckFeeRate, relayMinFee, DEFAULT_MIN_RELAY_TX_FEE,
+mempoolMinFee, and "mempool min fee"
+- Look at src/validation.cpp::517 CheckFeeRate that was added in the ATMP
+  refactoring PR
+- Class member boolean `m_cpfpable` refers to the state of being CPFP-able,
+e.g. eligible for Child Pays For Parent
+
+For more background, here are [all the PRs by
+Suhas](https://github.com/bitcoin/bitcoin/pulls?q=is%3Apr+author%3Asdaftuar).
 
 ## Questions
 
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or
+NACK?](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)
+(Don't forget to put your PR review on GitHub.)
+
+2. What steps did you take, beyond reading the code and running the tests?
+
+3. Did you review the functional test, or address the author's TODO in it?
+
+4. Now that [PR #16400](https://github.com/bitcoin/bitcoin/pull/16400) has been
+merged, is this PR still relevant?
+
+Other questions you may want to consider:
+
+- Who is responsible for initiating package relay, sender or recipient?
+
+- Should we update the p2p protocol to accommodate package relay, or rely on
+existing functionality?
+
+- What Denial-of-Service concerns do we need to address?
+
+- One of the conditions in the PR is that *"No transactions in the mempool
+conflict with any transactions in the package."* What is meant here by
+conflict, and for what reasons could a parent transaction evict a transaction
+from the mempool that another child depends on?
+
+- From the PR: *"The ancestor/descendant size limits are calculated assuming
+that any mempool ancestor of any candidate transaction is an ancestor of all the
+candidate transactions."* Do you agree that this assumption can be made?
+
+## Going further
```

Comment:
Great addition!

Reply:
Maybe for next week if nothing else is scheduled? Could also see which PRs go to high-priority during the irc meeting Thursday.